data/tutorials/gs_01_a_first_hour_with_ocaml.md

```diff
@@ -70,32 +90,91 @@ $ ocaml
+or use an absolute path.
+
+### Using the OCaml Playground
+
+The in-browser [OCaml Playground](https://ocaml.org/play), which you can
+use without installing anything, has this interface:
+
+![OCaml Playground](/media/tutorials/ocaml-playground.png)
+
+There are two windows: the editor window (on the left) is where you write code,
+the output window (on the right) is where all the answers from OCaml appear.
+
+Instead of entering code examples line by line on a prompt, as in the toplevel,
+you type or copy code into the editor window. To run all code in the editor window,
+you click on the "Run" button at the bottom of the editor window.
 
-## Expressions
+When using the OCaml Playground, it is not required to end expressions with
+`;;`. In this sense, the OCaml Playground behaves similar to how you would
+write OCaml code in a file.
 
-Our phrase `50 * 50` was an expression that evaluated to `2500`. OCaml told
-us that the type was `int`, an integer. Every expression in OCaml has a type.
-To avoid repetition, we can give a name to our number:
+
+## Expressions and Variables
+
+```ocaml
+# 50 * 50;;
+- : int = 2500
+```
+
+In OCaml, expressions have types and `50 * 50` is an expression that evaluates to the `int` type value of `2500`.
+Other primitive types in OCaml include `float`, `string`, and `bool`.
+
+To avoid repetition, we can assign a name to a value using the `let` keyword:
 
 ```ocaml
 # let x = 50;;
 val x : int = 50
+
 # x * x;;
 - : int = 2500
 ```
 
-Note that this isn't a variable as in languages like C and Python. Its value
-cannot be changed.
+When we type `let x = 50;;` and Enter, the REPL responds with the variable's value (`val`),
+showing that `x` now means `50`. So `x * x;;` evaluates to the same answer as `50 * 50;;`.
+
+It's important to note that, unlike in imperative languages,
+variables in OCaml are *immutable*, meaning their value cannot
+change after assignment. Also, variable names must begin with
+a lowercase letter or an underscore, and they cannot contain dashes
+(e.g., `x-plus-y` is not a legal variable name, but `x_plus_y` is).
+
+
+We can combine a variable declaration with an expression using the `let` ... `in` ... syntax:
+
+```ocaml
+# let x = 50 in x * x;;
+- : int = 2500
+```
+
+This declares the variable `x` and assigns it the value `50`,
+  which is then used in the expression `x * x`, resulting in the value of `2500`.
+
+Note that `let` ... `in` ... and `let` ... are two fundamentally different things:
 
-We can write also write it all on one line using `let` ... `in` ...:
+* `let x = 50;;` is a *variable declaration*. It assigns the value `50` to the variable `x`,
+which is bound in the toplevel.
+* `let x = 50 in x * x;;` is an *expression that includes a variable declaration*: The variable `x`
+is only bound within the expression following the `in` keyword.
 
 ```ocaml
 # let x = 50 in x * x;;
 - : int = 2500
+
+# x;;
+Error: Unbound value x
```

Comment:
Here I get the same answer `- : int = 50`, not the Error

Reply:
Thank you, I think I have to use a different name for the variable, because `x` is still bound from the expressions that were evaluated beforehand.